oxtorrent.py

```diff
@@ -25,127 +26,107 @@
+                    'desc_link': torrents[torrent][5]
+                }
+                prettyPrinter(data)
+
+        def __findTorrents(self, html):
+            torrents = []
+            trs = re.findall(
+                r'<td .+i> <a href=\".+</a></div></td>\n<td .+\n<td .+\n<td .+</td>\n</tr>', html)
+            for tr in trs:
+                # Extract from the A node all the needed information
+                url_titles = re.search(
+                    r'<td .+i> <a href=\"(.+)\" title=.+>(.+)</a></div></td>\n<td .+\"left\">'
+                    '([0-9,.]+ (TB|GB|MB|KB))</td>\n<td .+\"seeders\">.*([0-9,.]+).*</td>\n'
+                    '<td .+\"leechers\">.*([0-9,.]+).*</td>\n</tr>', tr)
+                if url_titles:
+                    torrents.append(
+                        [self.url + url_titles.group(1), url_titles.group(2), url_titles.group(3),
+                         url_titles.group(5), url_titles.group(6), self.url + url_titles.group(1)])
+            return torrents
+
+    def download_torrent(self, info):
+        # we have to fetch the info page and extract the magnet link
+        info_page = retrieve_url(urllib.parse.unquote(info))
+        magnet_match = re.search(r"window.location.href.*=.*\'(magnet[^\"]+)\';", info_page)
+        if magnet_match and magnet_match.groups():
+            print(magnet_match.groups()[0] + " " + info)
+        else:
+            raise Exception('Error, please fill a bug report!')
 
+#    def download_torrent(self, info):
+#        info_page = retrieve_url(urllib.parse.unquote(info))
+#        file_link = re.search(r"window.location.href.*=.*\'(/telecharger/.+)\';", info_page)
+#        torrent_url = self.url+file_link.group(1)
+#        print(download_file(torrent_url))
 
+    # DO NOT CHANGE the name and parameters of this function
+    # This function will be the one called by nova2.py
     def search(self, what, cat='all'):
         """ Performs search """
+        """
+        Here you can do what you want to get the result from the search engine website.
+        Everytime you parse a result line, store it in a dictionary
+        and call the prettyPrint(your_dict) function.
+
+        `what` is a string with the search tokens, already escaped (e.g. "Ubuntu+Linux")
+        `cat` is the name of a search category in ('all', 'movies', 'tv', 'music', 'games',
+        'anime', 'software', 'pictures', 'books')
+        """
         #prepare query. 7 is filtering by seeders
         cat = cat.lower()
-        query = "/".join((self.url, "recherche", what))
-        response = retrieve_url(query)
-        list_searches = []
-        parser = self.MyHtmlParser(list_searches, self.url)
-        parser.feed(response)
+        what = what.replace("%20", "+")
+
+        parser = self.HTMLParser(self.url)
+        page = 1
+        while True:
+            page_url = "{0}/recherche/{1}/{2}".format(self.url, what, page)
+            print(page_url)
```

Comment:
Should it be removed?

Reply:
Oups…
I forget one debug, you are right !
Thanks,
I will fix this